src/System Application/App/Agent/Setup/AgentUserSubform.PageExt.al

```diff
@@ -0,0 +1,109 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Agents;
+
+using System.Security.User;
+using System.Security.AccessControl;
+
+pageextension 4318 "Agent User Subfrom" extends "User Subform"
+{
+    layout
+    {
+        modify(Company)
+        {
+            Visible = (not IsAgent) or ShowCompanyField;
+        }
+    }
+
+    actions
+    {
+        addlast(Processing)
+        {
+            action(AgentShowHideCompany)
+            {
+                ApplicationArea = All;
+                Caption = 'Show/hide company';
+                Enabled = IsAgent;
+                Image = CompanyInformation;
+                Visible = IsAgent;
+
+                trigger OnAction()
+                begin
+                    ShowCompanyOverride := true;
```

Comment:
We need a small confirm here explaining the consequences.

Reply:
I'll add that